Tests/kaas/k8s-version-policy/k8s_version_policy.py

```diff
@@ -0,0 +1,531 @@
+            if release.age > MINOR_VERSION_CADENCE:
+                return False
+
+        if my_version.branch == release.version.branch and my_version.patch < release.version.patch:
+            if release.age > PATCH_VERSION_CADENCE:
+                return False
+
+            if release.version in cve_version_list and release.age > CVE_VERSION_CADENCE:
+                return False
+
+        if release.version.minor == (my_version.minor + 1) and release.version.patch == 0:
+            break
+
+    return True
+
+
+def parse_branch_info(data: dict) -> K8sBranchInfo:
+    major, minor = data["branch"].split(".")
+    branch = K8sBranch(int(major), int(minor))
+    eol_date = datetime.strptime(data["end-of-life"], "%Y-%m-%d")
+    return K8sBranchInfo(branch, eol_date)
+
+
+def read_supported_k8s_branches(eol_data_path: Path) -> dict[K8sBranch, K8sBranchInfo]:
+    with open(eol_data_path) as stream:
+        data = yaml.load(stream, Loader=yaml.FullLoader)
+    infos = [parse_branch_info(item) for item in data]
+    return {info.branch: info for info in infos}
+
+
+async def main(argv):
+    try:
+        config = initialize_config(parse_arguments(argv))
+    except (OSError, ConfigException, HelpException) as e:
+        if hasattr(e, 'message'):
+            logger.error(e.message)
+        print_usage()
+        return 1
+
+    connector = aiohttp.TCPConnector(limit=5)
+    async with aiohttp.ClientSession(connector=connector) as session:
+        cve_affected_ranges = await collect_cve_versions(session)
+
+    contexts = ["stable", "oldstable", "oldoldstable", "oldoldoldstable"]
+    branch_infos = read_supported_k8s_branches(Path(HERE, "k8s-eol-data.yml"))
+    supported_branches = {
+        branch
+        for branch, branch_info
+        in branch_infos.items()
+        if branch_info.is_supported()
+    }
+    seen_branches = set()
+
+    for context in contexts:
+        logger.info("Checking cluster of kubeconfig context '%s'.", context)
+        try:
+            cluster = await get_k8s_cluster_info(config.kubeconfig, context)
+        except KubeConfigException as e:
+            logger.error("There was an error while connecting to the cluster: %s", e)
+            return 1
+        cluster_branch = cluster.version.branch
+        seen_branches.add(cluster_branch)
+
+        # allow older k8s branches, but not for the first context (stable)
+        allow_older = context != contexts[0]
+
+        if check_k8s_version_recency(cluster.version, cve_affected_ranges, allow_older):
+            logger.info(
+                "The K8s cluster version %s of cluster '%s' is still in the recency time window.",
+                cluster.version,
+                cluster.name,
+            )
+        else:
+            logger.error(
+                "The K8s cluster version %s of cluster '%s' is outdated according to the standard.",
+                cluster.version,
+                cluster.name,
+            )
+            return 2
```

Comment:
It would be preferable to collect the problems for all contexts. Then the return code could be computed as the maximum, or we do away with these return codes and we just return the number of errors (like with the entropy check, for instance).

Reply:
I agree :+1: working on it.